cdap-runtime-ext-dataproc/src/test/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisionerTest.java

```diff
@@ -363,4 +365,43 @@ public void testClusterMarkedForReuseOnDelete() throws Exception {
                         addedLabelsCaptor.getValue().keySet());
   }
 
+  @Test
+  public void testVersionComparison() {
+    VersionInfo nullVersion = new DataprocImageVersion(null);
+    VersionInfo emptyVersion = new DataprocImageVersion("");
+    VersionInfo v0 = new DataprocImageVersion("0");
+    VersionInfo v1 = new DataprocImageVersion("1");
+    VersionInfo v1debian = new DataprocImageVersion("1-debian");
+    VersionInfo v1dot4 = new DataprocImageVersion("1.4");
+    VersionInfo v1dot4dot99 = new DataprocImageVersion("1.4.99");
+    VersionInfo v1dot5 = new DataprocImageVersion("1.5");
+    VersionInfo v1dot5debian = new DataprocImageVersion("1.5-debian");
+    VersionInfo v1dot5dot0 = new DataprocImageVersion("1.5.0");
+    VersionInfo v1dot5dot0debian = new DataprocImageVersion("1.5.0-debian");
+    VersionInfo v1dot5dot0dot0 = new DataprocImageVersion("1.5.0.0");
+    VersionInfo v1dot5dot1 = new DataprocImageVersion("1.5.1");
+    VersionInfo v1dot6 = new DataprocImageVersion("1.6");
+    VersionInfo v2 = new DataprocImageVersion("2");
+    VersionInfo v2debian = new DataprocImageVersion("2-debian");
+    VersionInfo v2dot0 = new DataprocImageVersion("2.0");
+
+    Assert.assertTrue(v1dot5.compareTo(nullVersion) > 0);
```

Comment:
Does it mean we will fail for null version? It's better not to fail as I am not sure what custom images would return